components/automate-cli/cmd/chef-automate/restart.go

```diff
@@ -49,6 +99,167 @@ func runRestartServices(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-func init() {
-	RootCmd.AddCommand(restartServicesCmd)
+type ResultWithError struct {
+	result interface{}
+	err    error
+}
+
+func runRestartFromBastion(flags *RestartCmdFlags, rs RemoteCmdExecutor, nu NodeOpUtils) error {
+	infra, _, err := nu.getHaInfraDetails()
+	if err != nil {
+		return err
+	}
+	if flags.timeout < DEFAULT_TIMEOUT_FOR_RESTART {
+		return errors.Errorf("The operation timeout duration for each individual node during the restart should be set to a value greater than %v seconds.", DEFAULT_TIMEOUT_FOR_RESTART)
+	}
+	if !flags.automate && !flags.chefServer && !flags.opensearch && !flags.postgresql {
+		if len(flags.node) != 0 {
+			return status.Errorf(status.InvalidCommandArgsError, "Please provide service flag")
+		}
+		flags.automate = true
+		flags.chefServer = true
+		flags.opensearch = true
+		flags.postgresql = true
+	}
+	errChan := make(chan *ResultWithError, 4)
+	runRestartCmdForFrontEnd(infra, flags, rs, errChan)
+
+	if nu.isManagedServicesOn() {
+		errChan <- &ResultWithError{}
+		errChan <- &ResultWithError{}
```

Comment:
Why this statement is 2 times ?

Reply:
For each node type there's a channel Incase of managed services if I dont make two remaining node's(os&pg) channel as nil it will wait for it